Sprint-3/alarmclock/alarmclock.js

```diff
@@ -1,25 +1,101 @@
-function setAlarm() {}
+const audio = new Audio("alarmsound.mp3");
+let countdownInterval;
+let remainingTime = 0;
 
-// DO NOT EDIT BELOW HERE
+function formatTime(seconds) {
+  const minutes = Math.floor(seconds / 60);
+  const secs = seconds % 60;
+  return `${String(minutes).padStart(2, "0")}:${String(secs).padStart(2, "0")}`;
+}
 
-var audio = new Audio("alarmsound.mp3");
+function updateDisplay(time) {
+  const timeRemaining = document.getElementById("timeRemaining");
+  timeRemaining.innerText = `Time Remaining: ${formatTime(time)}`;
+}
 
-function setup() {
-  document.getElementById("set").addEventListener("click", () => {
-    setAlarm();
-  });
+function setAlarm() {
+  const inputField = document.getElementById("alarmSet");
+  remainingTime = parseInt(inputField.value, 10);
+
+  if (isNaN(remainingTime) || remainingTime <= 0) {
```

Comment:
because you have the type of the input set as 'number' in the HTML, you actually don't need `isNaN` because it is impossible for the user to enter anything other than a number